frontend/src/scenes/instance/AsyncMigrations/asyncMigrationsLogic.ts

```diff
@@ -149,19 +157,70 @@ export const asyncMigrationsLogic = kea<asyncMigrationsLogicType>({
                 },
             },
         ],
-    }),
+    })),
 
-    selectors: {
+    selectors({
         isAnyMigrationRunning: [
             (s) => [s.asyncMigrations],
             (asyncMigrations) =>
                 asyncMigrations.some((migration) =>
                     [AsyncMigrationStatus.Running, AsyncMigrationStatus.Starting].includes(migration.status)
                 ),
         ],
-    },
+        isFutureMigration: [
+            (s) => [s.preflight],
+            (preflight) =>
+                function isFutureMigration(migration: AsyncMigration): boolean {
+                    const posthogVersion = preflight?.posthog_version
+                    return !posthogVersion || isVersionLT(posthogVersion, migration.posthog_min_version)
```

Comment:
We're spreading business logic here across too many places.

Could we move all business logic (what's actionable, what's future, version checking) to the backend and keep it in one place?